### ISSUE TITLE
RFC 3339 + Python

### ISSUE THREAD

**Issue by pjmagee** [microsoft/kiota-serialization-json-python#361]
### What are you generating using Kiota, clients or plugins?

API Client/SDK

### In what context or format are you using Kiota?

Nuget tool

### Client library/SDK language

Python

### Describe the bug

Python Serialized date: 2024-09-11T03:33:16+00:00**Z**

Link to Microsoft ISO/RFC 3339  for .NET:
```
https://learn.microsoft.com/en-us/dotnet/standard/datetime/system-text-json-support#the-extended-iso-8601-12019-profile-in-systemtextjson
```
When the Python SDK sends a request to an API using .NET `System.Text.Json` to Parse and validate RFC 3339 from a Kiota Python SDK datetime, it cannot deserialize the datetime. 

If the Python code omitted the Z at the end, it would be valid because including the `+00:00` timezone already means its UTC. 

From the Microsoft page on RFC Parsing (for .NET at least)

```
4. "'Full date''T''Time hour'':''Minute''Time offset'"
"yyyy'-'MM'-'dd'T'HH':'mmZ"
"yyyy'-'MM'-'dd'T'HH':'mm('+'/'-')HH':'mm"
```

Check the link on RFC 3339 section 5.6:

https://datatracker.ietf.org/doc/html/rfc3339#section-5.6

It states `Z OR time-numoffset`

### Expected behavior

`2024-09-11T03:33:16+00:00Z` should be serialised as `2024-09-11T03:33:16+00:00`

### How to reproduce

Any openapi spec with a `datetime` sent to a server to parse RFC 3339 datetime. 

Using .NET System.Text.Json RFC 3339 Support

### Kiota Version

1.18.0

### Debug output

<details><summary>Click to expand log</summary>
```


<log output here>

```
</details>

**Comment by baywet** [comment]
Hi @pjmagee 
Thank you for using kiota and for reaching out.

Let's use RFC3339 moving forward in this discussion since [it's the standard that OAS uses](https://spec.openapis.org/registry/format/date-time.html)

If I understand your issue properly:
- When serializing a DateTimeOffset in dotnet which is on the UTC standard you get +00:00 or Z at the end, but not both
- When serializing a datetime in pythonwhich is on the UTC standard you get +00:00Z at the end, which is incorrect according to RFC3339

Does that reflect the situation properly?

If so, I've transferred this issue to the Python serialization repository, this is [most likely caused by this block](https://github.com/microsoft/kiota-serialization-json-python/blob/01d8512d91e791e726481bc08417f5b5a86dc788/kiota_serialization_json/json_serialization_writer.py#L108)

Unit tests are implemented [here](https://github.com/microsoft/kiota-serialization-json-python/blob/01d8512d91e791e726481bc08417f5b5a86dc788/tests/unit/test_json_serialization_writer.py#L98) (but only for the string type).

Is this something you'd like to submit a pull request for provided some guidance?

**Comment by pjmagee** [comment]
> Hi @pjmagee Thank you for using kiota and for reaching out.
> 
> Let's use RFC3339 moving forward in this discussion since [it's the standard that OAS uses](https://spec.openapis.org/registry/format/date-time.html)
> 
> If I understand your issue properly:
> 
> * When serializing a DateTimeOffset in dotnet which is on the UTC standard you get +00:00 or Z at the end, but not both
> * When serializing a datetime in pythonwhich is on the UTC standard you get +00:00Z at the end, which is incorrect according to RFC3339
> 
> Does that reflect the situation properly?
> 
> If so, I've transferred this issue to the Python serialization repository, this is [most likely caused by this block](https://github.com/microsoft/kiota-serialization-json-python/blob/01d8512d91e791e726481bc08417f5b5a86dc788/kiota_serialization_json/json_serialization_writer.py#L108)
> 
> Unit tests are implemented [here](https://github.com/microsoft/kiota-serialization-json-python/blob/01d8512d91e791e726481bc08417f5b5a86dc788/tests/unit/test_json_serialization_writer.py#L98) (but only for the string type).
> 
> Is this something you'd like to submit a pull request for provided some guidance?

Yes, this seems to be what i find occurring. I also am using `System.Text.Json` to 'deserialise' RFC 3339 Dates as our server is .NET, but I am using Kiota 'examples' in various languages (go, python, c#) etc. 

I am happy to try and raise a PR, similar to the other date issue i fixed a while back.

**Comment by baywet** [comment]
Let us know if you have any additional comments or questions.

**Comment by baywet** [comment]
Thank you for the additional information.
I've seen multiple notifications come in, this is the first one I'm looking at, this answer might be a bit behind.

There are effectively multiple places where the format of a date time matters:
- URIs (path and query parameters)
- serialization writers (and parse node) text/json/multipart/form

As a general rule of thumb: kiota client should try to normalize URI parameters as much as it can before sending it down to std uri template. This is because we have more context and dependencies than this lib has and can handle more data types it can.

We should also strive to have the same implementation for date time across abstractions and serialization implementations to achieve maximal consistency.
